plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/AssignAnalysis.java

```diff
@@ -391,6 +391,13 @@ private TokensList getNonFunctionDefCompletionsFromAssign(ICodeCompletionASTMana
                 ICompletionState copy = state.getCopy();
                 if (definition.ast instanceof Attribute) {
                     copy.setActivationToken(NodeUtils.getFullRepresentationString(definition.ast));
+                } else if (definition.ast instanceof Assign) {
```

Comment:
Since you have the definition here and we put the typing info on it, I think that instead of checking the `definition.ast` for a special case, you could do:

```
if(definition.type != null && !definition.type.isEmpty()){
    copy.setActivationToken(definition.type)
}
```

Also, if there's a type comment it should have priority over the value (so, it should be checked first).